gtda/curves/features.py

```diff
@@ -0,0 +1,116 @@
+"""Feature extraction from curves."""
+
+from types import FunctionType
+
+from gtda.base import BaseEstimator
+from sklearn.base import TransformerMixin
+from sklearn.utils.validation import check_is_fitted, check_array
+
+from ._functions import _AVAILABLE_FUNCTIONS, _parallel_featurization
+from ..utils._docs import adapt_fit_transform_docs
+from ..utils.validation import validate_params
+
+
+@adapt_fit_transform_docs
+class StandardFeature(BaseEstimator, TransformerMixin):
+    """Computes standard features of multi-channels curves.
+
+    Given a multi curve applies any function to extract features from it.
+
+    Parameters
+    ----------
+    function : str or callable, optional, default: ``max``
+        Function to transform a multi channel curve into features. Implemented
+        functions are [``'argmin'``, ``'argmax'``, ``'min'``, ``'max'``,
+        ``'mean'``, ``'std'``, ``'median'``, ``'average'``]
```

Comment:
So, in general, suppose I have a multichannel curve `S`, is f applied on each channel of S, or on S as a whole?

Reply:
I tried to make that a bit more clear. Let me know what you think.